packages/tracing/src/transaction.ts

```diff
@@ -88,6 +97,15 @@ export class Transaction extends SpanClass {
       }).endTimestamp;
     }
 
+    const extra: Partial<Event> = {};
```

Comment:
If I understand this well, `extra` here is an unfortunate name because it confuses with the `event.extra` attribute.

Suggestion: initialize an event explicitly and set measurements conditionally before calling `captureEvent`. Alternatively, always set `event.measurements` unconditionally and then have the event `toJSON` drop the undefined keys (it is already implemented, grep for `dropUndefinedKeys`).

Reply:
Done in https://github.com/getsentry/sentry-javascript/pull/2909/commits/8cc49f8d3c3655b0f1f230f43c56bb256185e35f